### ISSUE TITLE
[Public Catalog] Use Gatsby SSR API to query NerdGraph for quickstart details

### ISSUE THREAD

**Issue by zstix** [newrelic/developer-website#1854]
## Description
In addition to [updating the I/O landing page](https://github.com/newrelic/developer-website/issues/1853), we should also update the quickstart details page to take advantage of Gatsby's ["server-side" rendering API](https://v4.gatsbyjs.com/docs/reference/rendering-options/server-side-rendering/) to query NerdGraph. As with the landing page, we should export a new async function (`getServerData()`) and use the new prop (`serverData`) in [our quickstart details template](https://github.com/newrelic/developer-website/blob/develop/src/templates/QuickstartDetails.js).

#### NerdGraph
We _should_ be able to use the `nr1Catalog` namespace in NerdGraph. If not, we can always use the `openInstallation` namespace. The details for quickstart components (alerts, dashboards, ect.) appear to be stitched in. While it's not 100% complete, here's the start for a query to fetch that information:

```graphql
{
  actor {
    nr1Catalog {
      quickstarts {
        results {
          id
          metadata {
            displayName
            icon {
              url
            }
            quickstartComponents {
              ... on Nr1CatalogQuickstartDashboard {
                metadata {
                  previews {
                    url
                  }
                }
              }
              metadata {
                description
                displayName
              }
            }
          }
        }
      }
    }
  }
}

```

## Acceptance Criteria
* [x] https://github.com/newrelic/developer-website/issues/1821 (proof-of-concept & Gatsby upgrade)
* [x] https://github.com/newrelic/developer-website/issues/1852 (feature branch & Gatsby Cloud)
* [ ] Create a query to NerdGraph to get the information needed for the quickstart details
* [ ] Update the [quickstart details template](https://github.com/newrelic/developer-website/blob/develop/src/templates/QuickstartDetails.js) to load information from the API results (not the JSON)
* [ ] Display a loading state while the information is being fetched (if necessary)

**Comment by jpvajda** [comment]
DevEx is working on adding several fields, that aren't currently available in the `nr1-catalog` namespace, that will need to be available for us to finish this work, @tyreer is the captain of that epic.

**Comment by aswanson-nr** [comment]
Merged #1930 into the feature branch.
